Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/CarouselViewGalleries/CollectionCarouselViewGallery.cs

```diff
@@ -0,0 +1,208 @@
+
+			var newObservableButton = new Button
+			{
+				Text = "Set new ObservableCollection"
+			};
+
+			newObservableButton.SetBinding(Button.CommandProperty, "NewObservableCommand");
+
+			var addObservableButton = new Button
+			{
+				Text = "Add new Items to ObservableCollection"
+			};
+
+			addObservableButton.SetBinding(Button.CommandProperty, "NewItemsObservableCommand");
+
+			var threadObservableButton = new Button
+			{
+				Text = "Use ObservableCollection (Threads)"
+			};
+
+			threadObservableButton.SetBinding(Button.CommandProperty, "TheadCommand");
+
+			stack.Children.Add(clearButton);
+			stack.Children.Add(newObservableButton);
+			stack.Children.Add(addObservableButton);
+			stack.Children.Add(threadObservableButton);
+
+			layout.Children.Add(carouselView, 0, 0);
+			layout.Children.Add(scroll, 0, 1);
+
+   			Content = layout;
+		}
+
+		internal DataTemplate GetCarouselTemplate()
+		{
+			return new DataTemplate(() =>
+			{
+				var grid = new Grid();
+
+				var info = new Label
+				{
+					HorizontalOptions = LayoutOptions.Center,
+					VerticalOptions = LayoutOptions.Center,
+					Margin = new Thickness(6)
+				};
+
+				info.SetBinding(Label.TextProperty, new Binding("Name"));
+
+				grid.Children.Add(info);
+
+				var frame = new Frame
+				{
+					Content = grid,
+					HasShadow = false
+				};
+
+				frame.SetBinding(BackgroundColorProperty, new Binding("Color"));
+
+				return frame;
+			});
+		}
+	}
+
+	[Preserve(AllMembers = true)]
+	public class CollectionCarouselViewGalleryViewModel : BindableObject
+	{
+		readonly Random _random;
+		ObservableCollection<CarouselData> _items;
+
+		public CollectionCarouselViewGalleryViewModel()
+		{
+			_random = new Random();
+		}
+
+		public ObservableCollection<CarouselData> Items
+		{
+			get { return _items; }
+			set
+			{
+				_items = value;
+				OnPropertyChanged();
+			}
+		}
+  
+		public ICommand ClearCommand => new Command(Clear);
+		public ICommand NewObservableCommand => new Command(NewObservable);
+		public ICommand NewItemsObservableCommand => new Command(NewItemsObservable);
+		public ICommand TheadCommand => new Command(async () => await Thead());
+
+		void Clear()
+		{
+			Items?.Clear();
+		}
+		  
+		void NewObservable()
+		{
+			Clear();
+
+			var items = new List<CarouselData>();
+
+			for (int n = 0; n < 5; n++)
+			{
+				items.Add(new CarouselData
+				{
+					Color = Color.FromRgb(_random.Next(0, 255), _random.Next(0, 255), _random.Next(0, 255)),
+					Name = $"NewObservable {n + 1}"
+				});
+			}
+
+			Items = new ObservableCollection<CarouselData>(items);
+		}
+  
+		void NewItemsObservable()
+		{
+			Clear();
+
+			Items = new ObservableCollection<CarouselData>();
+
+			for (int n = 0; n < 5; n++)
+			{
+				Items.Add(new CarouselData
+				{
+					Color = Color.FromRgb(_random.Next(0, 255), _random.Next(0, 255), _random.Next(0, 255)),
+					Name = $"NewItemsObservable {n + 1}"
+				});
+			}
+		}
+
+		async Task Thead()
```

Comment:
Thead?

Reply:
Yeah, please change this to `Thread` or something more descriptive.

Reply:
Oops, fixed!.